base/linalg/conjarray.jl

```diff
@@ -0,0 +1,45 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+"""
+    ConjArray(array)
+
+A lazy-view wrapper of an `AbstractArray`, taking the elementwise complex
+conjugate. This type is usually constructed (and unwrapped) via the `conj()`
+function (or related `ctranspose()`), but currently this is the default behavior
+for `RowVector` only.
+"""
+immutable ConjArray{T, N, A <: AbstractArray} <: AbstractArray{T, N}
+    parent::A
+end
+
+@inline ConjArray{T,N}(a::AbstractArray{T,N}) = ConjArray{conj_type(T), N, typeof(a)}(a)
+
+# This type can cause the element type to change under conjugation - e.g. an array of complex arrays.
+@inline conj_type(x) = conj_type(typeof(x))
+@inline conj_type{T}(::Type{T}) = promote_op(conj, T)
+
+@inline parent(c::ConjArray) = c.parent
+@inline parent_type(c::ConjArray) = parent_type(typeof(c))
+@inline parent_type{T,N,A}(::Type{ConjArray{T,N,A}}) = A
+
+@inline size(a::ConjArray) = size(a.parent)
+linearindexing{CA <: ConjArray}(::Union{CA,Type{CA}}) = linearindexing(parent_type(CA))
+
+@propagate_inbounds getindex{T,N}(a::ConjArray{T,N}, i::Int) = conj(getindex(a.parent, i))
+@propagate_inbounds getindex{T,N}(a::ConjArray{T,N}, i::Vararg{Int,N}) = conj(getindex(a.parent, i...))
+@propagate_inbounds setindex!{T,N}(a::ConjArray{T,N}, v, i::Int) = setindex!(a.parent, conj(v), i)
+@propagate_inbounds setindex!{T,N}(a::ConjArray{T,N}, v, i::Vararg{Int,N}) = setindex!(a.parent, conj(v), i...)
```

Comment:
Actually, not `pointer` since the underlying data is not conjugated.  But we should provide `pointer` for `RowVector`.  And we should still provide `stride` for `ConjArray` (this is potentially useful for telling you what direction to traverse an array).